cmd/node/main.go

```diff
@@ -634,32 +685,149 @@ func startNode(ctx *cli.Context, log *logger.Logger, version string) error {
 	return nil
 }
 
-func startStatusPolling(ash core.AppStatusHandler, pollingInterval int, networkComponents *factory.Network) error {
+func startStatusPolling(
+	ash core.AppStatusHandler,
+	pollingInterval int,
+	networkComponents *factory.Network,
+	processComponents *factory.Process,
+) error {
+
 	if ash == nil {
 		return errors.New("nil AppStatusHandler")
 	}
 
-	pollingIntervalDuration := time.Duration(pollingInterval) * time.Second
-	appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(ash, pollingIntervalDuration)
+	appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(ash, pollingInterval)
 	if err != nil {
 		return errors.New("cannot init AppStatusPolling")
 	}
 
+	err = registerPollConnectedPeers(appStatusPollingHandler, networkComponents)
+	if err != nil {
+		return err
+	}
+
+	err = registerPollProbableHighestNonce(appStatusPollingHandler, processComponents)
+	if err != nil {
+		return err
+	}
+
+	appStatusPollingHandler.Poll()
```

Comment:
The object `appStatusPollingHandler` will go out of scope at the end of the current function. But that's okay, because it has started all the necessary go routines, right?

Reply:
yes